src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -572,5 +601,27 @@ public void Dispose()
                 }
             }
         }
+
+        private static string GetMSbuild(string msbuildDirectory)
```

Comment:
Yes, please look it up in PATH.

It's not guaranteed that msbuild lives in /Library/Frameworks/Mono.framework/Versions/Current/lib/mono/msbuild/14.1/bin/, for example if you install mono via homebrew it might live in /usr/local/Cellar/mono instead.

And mono already adds msbuild to the PATH, plus it will solve the 14.1 vs 15 versioning issue.